### PR TITLE
added agnosticBinarySearch algorithm and  Testcases for ascending and descending order

### PR DESCRIPTION
**Pull request by HumbleFool830**
- [ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [ ] All filenames are in PascalCase.
- [ ] All functions and variable names follow Java naming conventions.
- [ ] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.

### CODE REVIEW

**Review by siriak — CHANGES_REQUESTED** [review]
Please fix formatting and name variables descriptively

**Comment by HumbleFool830** [comment]
@siriak can you check this once is there any modifications let me know !

File: src/main/java/com/thealgorithms/searches/agnosticBinarySearch.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        for(int i=0;i<n;i++){
+            arr[i]=sc.nextInt();
+        }
+        System.out.println("enter key element to search in a array");
+        int key=sc.nextInt();
+        sc.close();
+        int keyIndexPosition =agnosticbinarySearch(arr,key);
+        System.out.println(keyIndexPosition);        
+        
+    }
```
Comment (siriak) [inline src/main/java/com/thealgorithms/searches/agnosticBinarySearch.java:19]:
```suggestion
```

File: src/main/java/com/thealgorithms/searches/agnosticBinarySearch.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package com.thealgorithms.searches;
+import java.util.*;
+public class agnosticBinarySearch {
+    public static int agnosticbinarySearch(int arr[],int key){
```
Comment (siriak) [inline src/main/java/com/thealgorithms/searches/agnosticBinarySearch.java:4]:
```suggestion
public class sortOrderAgnosticBinarySearch {
    public static int find(int arr[], int key) {
```
The class name must say what it's agnostic to. The function name doesn't need to repeat what's already in the class name

File: src/test/java/com/thealgorithms/searches/agnosticBinarySearchTest.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
+package com.thealgorithms.searches;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.api.Test;
+
+public class agnosticBinarySearchTest{
+
+    @Test
+    public void Testascending(){
```
Comment (siriak) [inline src/test/java/com/thealgorithms/searches/agnosticBinarySearchTest.java:10]:
```suggestion
    public void testAscending(){
```
Please use camelCase for function names

**Review by siriak — CHANGES_REQUESTED** [review]

File: src/main/java/com/thealgorithms/searches/agnosticBinarySearch.java (hunk 2 above)
Comment (HumbleFool830) [reply src/main/java/com/thealgorithms/searches/agnosticBinarySearch.java]: @siriak changes done can check it once

Comment (HumbleFool830) [reply src/main/java/com/thealgorithms/searches/agnosticBinarySearch.java]: @siriak changes will cleared can you check is there any modifications let me know

File: src/main/java/com/thealgorithms/searches/sortOrderAgnosticBinarySearch.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        System.out.println("enter the "+n+" elements to store in an array");
+        for(int i=0;i<n;i++){
+            arr[i]=sc.nextInt();
+        }
+        System.out.println("enter key element to search in a array");
+        int key=sc.nextInt();
+        sc.close();
+        System.out.println(find(arr, key));     // function calling     
+        
+    }
```
Comment (siriak) [inline src/main/java/com/thealgorithms/searches/sortOrderAgnosticBinarySearch.java:18]:
```suggestion
```

**Review by siriak — APPROVED** [review]
Looks good, thanks!

**Pull request merged**